quarkus-workshop-super-heroes/super-heroes/ui-super-heroes/src/main/webui/src/app/fight/fight.component.html

```diff
@@ -21,14 +21,21 @@ <h2><a data-toggle="collapse" href="#heroPowers" role="button" aria-expanded="fa
     <div class="col-xs-6 col-sm-4 col-md-4">
         <div class="card-pf">
             <div class="card-pf-body">
-                <button (click)="newFighters()" class="btn btn-primary btn-block btn-lg"><h4><i
-                    class="fas fa-random"></i> NEW FIGHTERS</h4></button>
-                <button (click)="fight()" class="btn btn-danger btn-block btn-lg"><h4><i class="fab fa-battle-net"></i>
-                    FIGHT !</h4></button>
+                <button (click)="newFighters()" class="btn btn-primary btn-block btn-lg">
+                    <h4><i class="fas fa-random"></i> NEW FIGHTERS</h4>
+                </button>
+                <button (click)="fight()" class="btn btn-danger btn-block btn-lg">
+                    <h4><i class="fab fa-battle-net"></i> FIGHT !</h4>
+                </button>
             </div>
 
-            <div *ngIf="winner" class="winner-text">Winner is <span
-                [className]="winner == fighters.villain.name ? 'winner-villain' : 'winner-hero'">{{winner}}</span></div>
+            <div *ngIf="winner" class="winner-text">
+                Winner is <span [className]="winner == fighters.villain.name ? 'winner-villain' : 'winner-hero'">{{winner}}</span>
+                <button (click)="narrate()" class="btn btn-secondary btn-block btn-lg">
+                    <h4><i class="fab fa-comment"></i> NARRATE THE FIGHT</h4>
+                </button>
+                <div *ngIf="narration" class="narration-text">{{narration.toString()}}</div>
```

Comment:
Try `narration.narration`. The toString() method in Javascript is *nothing* like as useful as in Java.